openhands/runtime/base.py

```diff
@@ -224,20 +226,45 @@ async def _handle_action(self, event: Action) -> None:
                     gh_client = GithubServiceImpl(
                         external_auth_id=self.user_id, external_token_manager=True
                     )
+                    logger.info(f'Fetching latest github token for runtime: {self.sid}')
                     token = await gh_client.get_latest_token()
-                    if token:
-                        export_cmd = CmdRunAction(
-                            f"export GITHUB_TOKEN='{token.get_secret_value()}'"
+                    if not token:
+                        logger.info(
+                            f'Failed to refresh github token for runtime: {self.sid}'
                         )
 
+                    if token:
+                        raw_token = token.get_secret_value()
+
+                        if not self.prev_token:
+                            logger.info(
+                                f'Setting github token in runtime: {self.sid}\nToken value: {raw_token[0:5]}; length: {len(raw_token)}'
+                            )
+
+                        elif self.prev_token.get_secret_value() != raw_token:
+                            logger.info(
+                                f'Setting [NEW] github token in runtime {self.sid}\nToken value: {raw_token[0:5]}; length: {len(raw_token)}'
```

Comment:
```suggestion
                                f'Setting new github token in runtime {self.sid}\nToken value: {raw_token[0:5]}; length: {len(raw_token)}'
```
no need to highlight here I think--this is expected, and we can grep as-is